feat(condo): DOMA-8365 changed card styles, added checkbox card

nomerdvadcatpyat opened the pull request:
Default card:
![изображение](https://github.com/open-condo-software/condo/assets/52532264/d23f968d-319f-4ce4-9413-b100dcd62d33)

Card active state:
![изображение](https://github.com/open-condo-software/condo/assets/52532264/50374646-0f6b-46f4-9a48-39e00ac520bb)

Button card with full content:
![изображение](https://github.com/open-condo-software/condo/assets/52532264/a8cf07df-96ba-4632-ae8b-553f5a12a228)

Checkbox card with full content:
![изображение](https://github.com/open-condo-software/condo/assets/52532264/4a3d14ef-4a0d-4289-ab11-5f2df1faa39e)

Checkbox card active state:
![изображение](https://github.com/open-condo-software/condo/assets/52532264/3d194cc8-9077-4baf-b8fb-556c4adc333b)

SavelevMatthew reviewed (commented):
1. Checkbox hover behaviour is independent from card one. 
2. Checkbox card itself have no hover behaviour. 
3. WTF is black card about? I thought we're building light interfaces

nomerdvadcatpyat commented:
> 1. Checkbox hover behaviour is independent from card one.

Fixed. Added `pointer-events: none` to inner checkbox

>  2. Checkbox card itself have no hover behaviour.

Fixed. Added default `hoverable` to `CheckboxCard`, omit `hoverable` from `CheckboxCard` props

>  3. WTF is black card about? I thought we're building light interfaces

It's [new active state of card](https://www.figma.com/file/Nbkug4IRM33sEyDuN0yO1C/UI-Kit-(web)?type=design&node-id=8732-4676&mode=design&t=r5VgJdYuQ6Akn0lP-0)
For example, [active step in onboarding](https://www.figma.com/file/NnpWfYUo4Z8LGV3gHXRnHe/%D0%9E%D0%BD%D0%B1%D0%BE%D1%80%D0%B4%D0%B8%D0%BD%D0%B3?type=design&node-id=2301-17924&mode=design&t=W4Y1GoxovV6bgAqu-0)

Alllex202 commented:
1) Should it be possible to add a button on a card with a checkbox?
2) Should the state of checkbox change when button is clicked?

https://github.com/open-condo-software/condo/assets/56914444/677d1d28-4d03-4717-aedc-4907a17ae063

Alllex202 commented:
Something's not working here

<img width="1512" alt="Снимок экрана 2024-02-13 в 23 18 04" src="https://github.com/open-condo-software/condo/assets/56914444/ce32b034-6a44-4d87-9054-11b59c7ff5ed">
<img width="1511" alt="Снимок экрана 2024-02-13 в 23 17 56" src="https://github.com/open-condo-software/condo/assets/56914444/39812de3-23b5-4095-b6c2-50a8890eeeb0">

Alllex202 commented:
Perhaps we should merge these stories under one section “Card”

<img width="213" alt="Снимок экрана 2024-02-13 в 23 38 04" src="https://github.com/open-condo-software/condo/assets/56914444/2e7ac9ea-b0b9-4f0b-a4d6-d23a06d5e21a">

Alllex202 commented on packages/ui/src/components/Card/_utils/cardBody.less:
maybe such a structure? Like in `Dropdown`

Card
-- body
---- .tsx, .less
-- header
---- .tsx, .less
-- button
---- .tsx, .less
-- checkbox
---- .tsx, .less

Alllex202 commented on packages/ui/src/components/Card/_utils/utils.tsx:
`.../_utils/utils...`
Why the double name? It is already clear that this is a folder with utilities

Alllex202 commented on packages/ui/src/components/Card/_utils/utils.tsx:
```diff
@@ -0,0 +1,26 @@
+    return (
+        <Space size={8} direction='horizontal' align='center'>
+            {PreIcon && <PreIcon size='small' />}
+            <Typography.Link href={href}>{label}</Typography.Link>
```
Will this link be able to work as an SPA link (without reloading the page)? Or should it be like a default link?

nomerdvadcatpyat replied:
By default it's be like default link.
Added `LinkWrapper` prop to make it possible to be spa link (wrap in next/link from outside)

Alllex202 commented on packages/ui/src/components/Card/cardButton.tsx:
```diff
@@ -0,0 +1,45 @@
+
+
+export type CardButtonProps = Pick<CardProps, 'accent' | 'disabled'> & {
+    header: CardHeaderProps
```
Why can't we set a body without a header?

nomerdvadcatpyat replied:
Made header optional

nomerdvadcatpyat commented:
> 1. Should it be possible to add a button on a card with a checkbox?
> 
> 2. Should the state of checkbox change when button is clicked?
> 
> 
> 2024-02-13.23.32.16.mov

1. Yes
2. No

nomerdvadcatpyat commented on packages/ui/src/components/Emoji/Emoji.tsx:
I don’t think it’s necessary yet

Alllex202 commented on packages/ui/src/components/Card/checkbox/cardCheckbox.tsx:
```diff
@@ -0,0 +1,89 @@
+        setChecked(prev => !prev)
+
+        if (props.onClick) {
+            props.onClick(event)
```
I think here we should return the value of the checkbox

Alllex202 commented on packages/ui/src/components/Card/checkbox/cardCheckbox.tsx:
```diff
@@ -0,0 +1,89 @@
+        ...rest
+    } = props
+
+    const [checked, setChecked] = useState<boolean>()
```
what about default value?

Alllex202 commented on packages/ui/src/components/Card/checkbox/cardCheckbox.tsx:
```diff
@@ -0,0 +1,89 @@
+        if (props.onClick) {
+            props.onClick(event)
+        }
+    }, [props])
```
We don't need to track all props

Alllex202 commented on packages/ui/src/stories/Card/CardCheckbox.stories.tsx:
```diff
@@ -0,0 +1,94 @@
+import { Meta, Story } from '@storybook/react'
+import {  pickBy } from 'lodash'
```
```suggestion
import { pickBy } from 'lodash'
```

Alllex202 commented on packages/ui/src/stories/ProgressIndicator.stories.tsx:
I think you should send this component for design review

Alllex202 commented:
<img width="409" alt="Снимок экрана 2024-02-15 в 01 33 04" src="https://github.com/open-condo-software/condo/assets/56914444/67ddbe4e-e8ba-406a-9952-3a763d2183c6">

Alllex202 commented:
Should the picture shrink so much on the sides?

<img width="409" alt="Снимок экрана 2024-02-15 в 01 33 04" src="https://github.com/open-condo-software/condo/assets/56914444/1c04a93f-a65e-4a80-9174-54a67f3a75c5">

Alllex202 commented:
Links also change the state of the checkbox. I don't think it should be like that

https://github.com/open-condo-software/condo/assets/56914444/243dde0a-30a9-4053-a693-a53676cc261d

Alllex202 commented:
Something is wrong with picture

https://github.com/open-condo-software/condo/assets/56914444/f4ad6238-929a-4ce4-88d9-44f4d34741d1
